bindings/pyroot/pythonizations/src/PyROOTModule.cxx

```diff
@@ -60,6 +61,69 @@ PyObject *RegisterExecutorAlias(PyObject * /*self*/, PyObject *args)
    Py_RETURN_NONE;
 }
 
+namespace Internal {
+/// \brief A PyObject wrapper to track reference counting of external objects
+///
+/// This wrapper can be useful in shared ownership scenarios when a C++ object
+/// is created on the Python side and there is no easy way to track its ownership
+/// on the C++ side. If multiple instances of this class are given the same
+/// Python proxy, they will increase/decrease its reference counting following
+/// Python rules and ensure proper destruction of the underlying C++ object when
+/// no other Python objects are referencing it.
+class PyObjRefCountWrapper final {
+   PyObject *fObject{nullptr};
+
+   void Reset(PyObject *object)
+   {
+      if (fObject) {
+         Py_DECREF(fObject);
+         fObject = nullptr;
+      }
+      if (object) {
+         Py_INCREF(object);
+         fObject = object;
+      }
+   }
+
+public:
+   PyObjRefCountWrapper(PyObject *object) { Reset(object); }
+
+   ~PyObjRefCountWrapper() { Reset(nullptr); }
+
+   PyObjRefCountWrapper(const PyObjRefCountWrapper &other) { Reset(other.fObject); }
+
+   PyObjRefCountWrapper &operator=(const PyObjRefCountWrapper &other)
+   {
+      Reset(other.fObject);
+      return *this;
+   }
+
+   PyObjRefCountWrapper(PyObjRefCountWrapper &&other)
+   {
+      Reset(other.fObject);
+      other.Reset(nullptr);
+   }
+
+   PyObjRefCountWrapper &operator=(PyObjRefCountWrapper &&other)
+   {
+      Reset(other.fObject);
+      other.Reset(nullptr);
+      return *this;
```

Comment:
```suggestion
      fObject = other.fObject);
      other.fObject = nullptr;
      return *this;
```
To avoid incrementing and decrementing the reference counter, making moving truly cheap (can also be done for the move constructor).